lib/base-compiler.js

```diff
@@ -55,6 +56,11 @@ Compile.prototype.newTempDir = function () {
 Compile.prototype.writeFile = Promise.denodeify(fs.writeFile);
 Compile.prototype.readFile = Promise.denodeify(fs.readFile);
 Compile.prototype.stat = Promise.denodeify(fs.stat);
+Compile.prototype.couldHaveOptOutput = function(options, version) {
+    //TODO(jared): a compiler meta data gatherer could improve this
```

Comment:
Right, definitely a good point. As below; it's tricky to know what the right thing to do is. As it is though anyone running a clang needs a bunch of changes to make it work right (e.g. Intel asm output). Might be nicer to detect the compiler type at startup with some heuristics and then apply "clang-like" and "gcc-like" etc features...